client/src/utils/axios-adapter.js

```diff
@@ -0,0 +1,63 @@
+import axios from 'axios';
+
+class RequestAdapter {
+    constructor(props = {}) {
+        const { baseURL = 'http://localhost:8055/items/todolist', ...rest } = props;
+        this.adapter = axios.create({
+            baseURL,
+            ...rest,
+        });
+
+        this.adapter.interceptors.request.use(this.interceptRequest);
+        this.adapter.interceptors.response.use(
+            this.interceptResponse,
+            this.handleError
+        );
+    }
+
+    async interceptRequest(config) {
+        return config;
+    }
+
+    async interceptResponse(response) {
+        return response.data.data;
+    }
+
+    handleError(error) {
+        console.error("API Error:", error);
+        throw error;
+    }
+
+    fetchTodo() {
+        return this.adapter.get("/");
+    }
+
+    addTodo(todo){
+        return this.adapter.post("/",{
+            status: "draft",
+            todo,
+            completed: false,
+            isEditing: false
+        }, {
+            headers: {
+                'Content-Type': 'application/json',
+            }
+        });
+    }
+
+    deleteTodo(id) {
+        return this.adapter.delete(`/${id}`);
+    }
+
+    editTask(task, id) {
+        return this.adapter.patch(`/${id}`, {
+            todo: task
+        }, {
+            headers: {
+                'Content-Type': 'application/json',
+            }
+        });
+    }
+}
```

Comment:
in `axios-interceptors` didnt accept use case method. just accept API METHOD in this function. 

you should add functional callback for this 

you can create an `service` folder inside of it contains `todo.service.js`

in `todo.service.js` you can create any service that contains only `todo` cases in this with all methods of API you defined in `axios-adapter`

Reply:
gamau gamau gamau